### PR TITLE
mkdir should always return a path

### LINKED ISSUE #94: mkdir(path; exists_ok) returns a bool, if it already exists
`mkdir` is supposed to return a path.
(I think this was the last PR i got into Base before julia 1.0)
This is useful for things like
`path=mkdir(joinpath("a", "b"))`

```julia
julia> mkdir(p"foo")
p"foo"

julia> mkdir(p"bar"; exist_ok=true)
p"bar"

julia> mkdir(p"bar"; exist_ok=true)
false
```

### PR DESCRIPTION
**Pull request by rofinn**
Closes #94

### CODE REVIEW

File: src/test.jl [hunk 1]
```diff
@@ -610,10 +610,10 @@ module TestPaths
         @testset "mkdir" begin
             garply = ps.root / "corge" / "grault" / "garply"
             @test_throws ErrorException mkdir(garply)
-            mkdir(garply; recursive=true)
+            @test mkdir(garply; recursive=true) == garply
```
Comment (mattBrzezinski) [inline src/test.jl:613]: Irrelevant to this PR.

But why is this test file in `src/`?

Comment (iamed2) [reply src/test.jl:613]: I believe it's here so other path types in downstream packages can test their compliance with the interface

Comment (mattBrzezinski) [reply src/test.jl:613]: Oh cool, good to know!

Comment (rofinn) [reply src/test.jl:613]: Yep, we use it AWSS3

**Pull request merged**